pkgs/tools/typesetting/tex/texlive/combine.nix

```diff
@@ -195,7 +209,6 @@ in (buildEnv {
       rm "$link"
       makeWrapper "$target" "$link" \
         --prefix PATH : "${gnused}/bin:${gnugrep}/bin:${coreutils}/bin:$out/bin:${perl}/bin" \
-        --prefix PERL5LIB : "$PERL5LIB" \
```

Comment:
while playing around with this, I stumbled upon the following behaviour:
```
✗ nix-build -A texlive.combined.scheme-full
/nix/store/swgky9ggf2i46sa4kvxvlxqdnq8p3x4h-texlive-combined-full-2022.20221227

ulli on pumpkin nixpkgs (954019f) [$?] 
➜ ./result/bin/fmtutil                     
Can't locate TeXLive/TLUtils.pm in @INC (you may need to install the TeXLive::TLUtils module) (@INC contains: /nix/store/6ljka8irx9j91h4xpy08qz24gkmmkq4w-texlive-combined-2022-texmfroot/tlpkg /nix/store/6ljka8irx9j91h4xpy08qz24gkmmkq4w-texlive-combined-2022-texmfroot/texmf-dist/scripts/texlive /nix/store/m6kc0wg6zii4bcw0fqxmclgy27ph09va-perl-5.36.0/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi /nix/store/m6kc0wg6zii4bcw0fqxmclgy27ph09va-perl-5.36.0/lib/perl5/site_perl/5.36.0 /nix/store/m6kc0wg6zii4bcw0fqxmclgy27ph09va-perl-5.36.0/lib/perl5/5.36.0/x86_64-linux-thread-multi /nix/store/m6kc0wg6zii4bcw0fqxmclgy27ph09va-perl-5.36.0/lib/perl5/5.36.0) at ./result/bin/fmtutil line 45.
BEGIN failed--compilation aborted at ./result/bin/fmtutil line 45.
```

What happens here is that `fmtutil` populates `@INC` by running `kpsewhich` from PATH. In this case this is the wrong system kpsewhich, which returns the wrong TEXMFROOT which doesn't include `/tlpkg` yet.
Is it reasonable for software to assume that it's (first) in PATH? Should we support the case where it isn't? (i.e. by prepending $out/bin to $PATH when wrapping (everything) or patching the invocations to use absolute paths)

Reply:
Oof, that's because `fmtutil` is not wrapped, because it was patched manually. But we don't need to patch it manually anymore, so this is easy to fix.

Reply:
Now `fmtutil` is wrapped properly and should work in all cases (unless you export a `TEXMFCNF` that points at an incompatible install, but would break a native TeX Live too).

Reply:
Unfortunately the newly added `tlmgr` doesn't work due to a similar reason. It fails to correctly populate the perl include path and hence doesn't find `TeXLive/TLConfig.pm`. It uses a slightly different mechanism than fmtutil -- it doesn't use `kpsewhich` from path, but (unsuccessfully) tries to find it relative to its own location. Fixing this might require some patching.

I checked execution of all the perl scripts to make sure removing `PERL5LIB` from the wrapper doesn't introduce any breakages and all the scripts that don't start because of missing perl modules also did so prior to this change (they are `ulqda`, `svn-multi`, `pdfannotextractor`, `kanji-fontmap-creator`, `ketcindy`, `ctan-o-mat`, `ctanupload`, `bibzbladd`, `bibmradd`, `bibdoiadd` and `bbl2bib`).

So merging this as is is fine by me, tlmgr can also be fixed later. It's close to useless without tlpdb anyway (while `--usermode` appears the work fine, `init-usertree` doesn't, so it can only be used with tlmgr usertrees that have been created manually or with other texlive installations).

Reply:
Oh, I didn't see `tlmgr` coming. I was grepping through the old scripts folder of scheme-full and didn't realise I was pulling new scripts in. The other perl scripts do now show weird `@INC` shenanigans, or at least not `TEXMFROOT` related (I don't remember at this point), rather they need the `buildPerlPackage` treatment. Presumably to become part or build on top of #225503. Thanks for checking carefully!

Since we already ship non-working scripts, I agree we can wait to fix `tlmgr` later.